File: nutrition-assistant-api/src/main/java/pl/polsl/wachowski/nutritionassistant/api/diary/entry/food/FoodEntryDetails.java
package pl.polsl.wachowski.nutritionassistant.api.diary.entry.food;

import lombok.Value;
import pl.polsl.wachowski.nutritionassistant.api.units.FoodMassUnit;
import pl.polsl.wachowski.nutritionassistant.api.food.NutrientDetails;

import java.math.BigDecimal;
import java.util.Set;

@Value
public class FoodEntryDetails {

    String name;
    String brand;
    FoodMassUnit massUnit;
    BigDecimal amount;
    Set<NutrientDetails> nutrients;
    Short position;

}

File: nutrition-assistant-front/src/app/model/diary/food-entry-details.ts
import { FoodUnit } from '../units/food-unit.enum';
import { NutrientBasicInfo } from '../food/nutrients/nutrient-basic-info';

export interface FoodEntryDetails {

    readonly name: string;
    readonly brandName: string;
    readonly amount: number;
    readonly unit: FoodUnit;
    readonly position: number;
    readonly nutrients: NutrientBasicInfo[];

}
